### PR TITLE
fix(phasor): adapt to resized canvas

### LINKED ISSUE #874: Incorrect page and canvas viewport after resizing the playground page
This is similar with #330, but affects the page container size.

### LINKED ISSUE #1220: Wrong edgeless shape offset on different `devicePixelRatio`
After switching browser `devicePixelRatio` (pressing `cmd` + `+/-` in Chrome), there is an offset between mouse position and the visual shape position when adding shape by drag.

### PR DESCRIPTION
**Pull request by alt1o**
Fixes #1220 
Fixes #874 

https://user-images.githubusercontent.com/21084335/220062787-a8816ba9-3213-4168-b877-b9e9f8d8a154.mov

### CODE REVIEW

**Comment by alt1o** [comment]
fix: #1220

File: packages/phasor/src/renderer.ts [hunk 1]
```diff
@@ -8,32 +11,48 @@ export class Renderer {
...
-    this.height = bbox.height;
-    this._centerX = this.width / 2;
-    this._centerY = this.height / 2;
+
+    this._initSize();
+    this.setCenter(this.width / 2, this.height / 2);
 
     this._loop();
+
+    this._disposeResizeListener = Signal.disposableListener(
```
Comment (doodlewind) [inline packages/phasor/src/renderer.ts:33]: Is it possible to listen on the resizing of the canvas element itself? This could be more precise if possible.

Comment (alt1o) [reply packages/phasor/src/renderer.ts:33]: I want to use window resizing event handing devicePixelRatio changing. But It seems also having some issues. I will change it to ResizeObserver.

**Review by doodlewind — APPROVED** [review]
LGTM

**Pull request merged**